code/modules/power/lighting.dm

```diff
@@ -146,9 +138,15 @@
 	plane = OBJ_PLANE
 	layer = OBJ_LAYER
 	stage = 1
-	fixture_type = "flamp"
+	fixture_type = /obj/machinery/light/flamp
 	sheets_refunded = 2
 
+/obj/machinery/light_construct/flamp/update_icon()
+	switch(stage)
+		if(1) icon_state = "flamp-construct-stage1"
+		if(2) icon_state = "flamp-construct-stage2"

```

Comment:
Please put the contents of an if/else/for check under the check itself, for readability.